pkg/plugins/ca/builtin/manager.go

```diff
@@ -0,0 +1,129 @@
+package builtin
+
+import (
+	"context"
+	"fmt"
+
+	core_model "github.com/Kong/kuma/pkg/core/resources/model"
+
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/pkg/errors"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	system_proto "github.com/Kong/kuma/api/system/v1alpha1"
+	core_ca "github.com/Kong/kuma/pkg/core/ca"
+	ca_issuer "github.com/Kong/kuma/pkg/core/ca/issuer"
+	core_system "github.com/Kong/kuma/pkg/core/resources/apis/system"
+	core_store "github.com/Kong/kuma/pkg/core/resources/store"
+	secret_manager "github.com/Kong/kuma/pkg/core/secrets/manager"
+)
+
+type builtinCaManager struct {
+	secretManager secret_manager.SecretManager
+}
+
+func NewBuiltinCaManager(secretManager secret_manager.SecretManager) core_ca.Manager {
+	return &builtinCaManager{
+		secretManager: secretManager,
+	}
+}
+
+var _ core_ca.Manager = &builtinCaManager{}
+
+func (b *builtinCaManager) Ensure(ctx context.Context, mesh string, backend mesh_proto.CertificateAuthorityBackend) error {
+	_, err := b.getCa(ctx, mesh, backend.Name)
+	if core_store.IsResourceNotFound(err) {
+		if err := b.create(ctx, mesh, backend.Name); err != nil {
+			return errors.Wrapf(err, "failed to create CA for mesh %q and backend %q", mesh, backend.Name)
+		}
+	} else {
```

Comment:
If you are talking about such code
```
func (b *builtinCaManager) Ensure(ctx context.Context, mesh string, backend mesh_proto.CertificateAuthorityBackend) error {
	_, err := b.getCa(ctx, mesh, backend.Name)
	if core_store.IsResourceNotFound(err) {
		if err := b.create(ctx, mesh, backend.Name); err != nil {
			return errors.Wrapf(err, "failed to create CA for mesh %q and backend %q", mesh, backend.Name)
		}
	}
	return err
}

```
then it's not the same since we will return an error if a resource is not found and we create it successfully.